src/braket/circuits/serialization.py

```diff
@@ -33,6 +34,26 @@ class QubitReferenceType(str, Enum):
     PHYSICAL = "PHYSICAL"
 
 
+class SerializableProgram(ABC):
+    @abstractmethod
+    def to_ir(
+        self,
+        ir_type: IRType = IRType.OPENQASM,
+    ) -> str:
+        """Serializes the program into an intermediate representation.
+
+        Args:
+            ir_type (IRType): The IRType to use for converting the program to its
+                IR representation. Defaults to IRType.OPENQASM.
+
+        Raises:
+            ValueError: If the supplied `ir_type` is not supported.
+
+        Returns:
+            str: A representation of the program in the `ir_type` format.
+        """
+
+
```

Comment:
We can no longer reference the `autoqasm` module from within the `devices` module, since we're using `devices` within `autoqasm` now. The only thing from `autoqasm` that we used in `devices` was the `Program.to_ir()` method, so I'm extracting that into an abstract base class here.

Reply:
Cool! it feels right that devices wouldn't depend on autoqasm, too